bot/commands/cmd_wiki.go

```diff
@@ -39,92 +57,178 @@ func (cm *CommandManager) commandWiki(
 		return
 	}
 
+	if !storage.WikiExists() {
+		return
+	}
+
+	wikiResult := readWiki(args)
+
+	if wikiResult.Err == errManyThreads {
+		cm.Discord.ChannelMessageSend(message.ChannelID, errManyThreads.Error()+"\nThe most similar funcs/callbacks are:\n* __**"+strings.Join(wikiResult.Threads, "**__\n* __**")+"**__")
+		return
+	} else if wikiResult.Err == errNoThreadFound {
+		cm.Discord.ChannelMessageSend(message.ChannelID, errNoThreadFound.Error())
+		return
+	} else if wikiResult.Err == errCouldntReadThread {
+		cm.Discord.ChannelMessageSend(message.ChannelID, errCouldntReadThread.Error())
+	} else if wikiResult.Err == nil {
+		cm.Discord.ChannelMessageSend(message.ChannelID, wikiResult.Thread)
+	} else {
+		err = wikiResult.Err
+	}
+
+	return false, err
+}
+
+func readWiki(article string) *WikiReturns {
+	files, err := getWikiFiles()
+	if err != nil {
+		return &WikiReturns{Err: err}
+	}
+
+	threadToRead := searchThread(files, article)
+
+	switch len(threadToRead) {
+	case 0:
+		return &WikiReturns{Err: errNoThreadFound}
+	case 1:
+		formatedInfo, readErr := readThread(threadToRead[0])
+		if readErr == errCouldntReadThread {
+			return &WikiReturns{Err: errCouldntReadThread}
+		}
+		return &WikiReturns{Err: nil, Thread: formatedInfo}
+	default:
+		var threads []string
+		for foo := range threadToRead {
+			threads = append(threads, threadName(threadToRead[foo]))
+		}
+		return &WikiReturns{Err: errManyThreads, Threads: threads}
+	}
+}
+
+func readThread(file string) (string, error) {
+	wikiFile, openErr := os.Open(filepath.Join(".", "wiki", "scripting", file))
+	if openErr != nil {
+		return "", errCouldntReadThread
+	}
+	defer wikiFile.Close()
+
+	wikiAsByte, readErr := ioutil.ReadAll(wikiFile)
+	if readErr != nil {
+		return "", errCouldntReadThread
+	}
+
+	output := blackfriday.Run(wikiAsByte)
+	html := bluemonday.UGCPolicy().SanitizeBytes(output)
+	fmt.Println(string(html))
+
+	doc, docErr := goquery.NewDocumentFromReader(strings.NewReader(string(html)))
+	if docErr != nil {
+		log.Fatal(docErr)

```

Comment:
This should return the error, potentially with `errors.Wrap(err, "some message...")`